kpi/serializers/v2/service_usage.py

```diff
@@ -180,7 +183,9 @@ def _get_per_asset_usage(self, user):
                 'date', 'xform', 'counter'
             )
             .filter(
-                xform__in=xforms,
+                # get submission counters for all deployed assets OR the null xform counter
+                Q(user__in=[kpi_user.id for kpi_user in self._users], xform=None) |
```

Comment:
Same logic for `self._users` as `asset_list` above. It seems to be used only to create this list.  
What about renaming `self._users` to `self._user_ids`  and get the ids directly when fetching all users from the organization.

```python
    def _get_per_asset_usage(self, user):
         self._user_ids = [user.pk]

        # Get the organization ID passed in from the query parameters
        organization_id = self.context['request'].query_params.get(
            'organization_id'
        )
        self._get_organization_details(organization_id)
        <snippet>

    def _get_organization_details(self, organization_id=None):
         <snippet>
        # If the user is in an organization, get all org users so we can query their total org usage
        self._user_ids = User.objects.values_list('pk', flat=True).filter(
            organizations_organization__id=organization_id
        )
```

BTW, since `organization_id` is only used to be passed to `_get_organization_details()`, I would move all its assignment logic inside `_get_organization_details()`.  `_get_per_asset_usage()` is super long and deserves to be splitted ;-)